src/rules/prefer-repository-shorthand.ts

```diff
@@ -0,0 +1,103 @@
+import type ESTree from "estree";
+import type { JSONLiteral } from "jsonc-eslint-parser/lib/parser/ast.js";
+
+import { createRule } from "../createRule.js";
+
+const isGitHubUrl = (url: string) =>
+	/^(?:git\+)?(?:ssh:\/\/git@|http?s:\/\/)?(?:www\.)?github\.com\/.*\/.*/.test(
+		url,
+	);
+
+const cleanGitHubUrl = (url: string) =>
+	url
+		.replace(
+			/^(?:git\+)?(?:ssh:\/\/git@|http?s:\/\/)?(?:www\.)?github\.com\//,
+			"",
+		)
+		.replace(/\.git$/, "");
+
+export default createRule({
+	create(context) {
+		return {
+			JSONProperty: (node) => {
+				const message =
+					"Use shorthand repository URL for GitHub repository";
+				if (
+					node.key.type === "JSONLiteral" &&
+					node.key.value === "repository"
+				) {
+					if (node.value.type === "JSONObjectExpression") {
+						const { properties } = node.value;
+						const typeProperty = properties.find(
```

Comment:
[Refactor] Just nitpicking a bit, there are 7 matches for `.type === "JSONLiteral"`. What do you think about making a shared helper with a name like `function isStringJSONLiteral`? And/or, making a functional helper like `createFindForStringJSONLiteral` / `findStringJSONLiteral`?

Vague starting thoughts...

```ts
const typeProperty = findStringJSONProperty(properties, "type");
```

```ts
const typeProperty = properties.find(createFindForJSONProperty("type"));
```

...or something vaguely in that direction? I haven't put too much thought into it 😄. And no worries if you don't like this, not blocking at all.